actix-http/src/h1/dispatcher.rs

```diff
@@ -867,13 +890,21 @@ where
     T: AsyncRead + Unpin,
 {
     let mut read_some = false;
+
     loop {
         let remaining = buf.capacity() - buf.len();
+
+        // If buf is full return but do not disconnect since
+        // there is more reading to be done
+        if buf.len() >= HW_BUFFER_SIZE {
+            return Ok(Some(false));
+        }
```

Comment:
In testing I've found that this check is all that's needed to keep memory usage in check; without the MaxBuf struct.